File: app/src/main/java/workshop1024/com/xproject/net/PublisherService.java
package workshop1024.com.xproject.net;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import workshop1024.com.xproject.model.publisher.Publisher;

public interface PublisherService {
    @GET("/publisherses/{typeId}")
    Call<List<Publisher>> getPublishersByType(@Path("typeId") String typeId);
}

File: main/src/main/java/workshop1024/com/xproject/main/publisher/data/source/remote/PublisherService.kt
package workshop1024.com.xproject.main.publisher.data.source.remote

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import workshop1024.com.xproject.main.publisher.data.Publisher

interface PublisherService {
    @GET("/publisherses/{contentId}")
    fun getPublishersByContentType(@Path("contentId") contentId: String): Call<List<Publisher>>

    @GET("/publisherses/{languageId}")
    fun getPublishersByLanguageType(@Path("languageId") languageId: String): Call<List<Publisher>>
}